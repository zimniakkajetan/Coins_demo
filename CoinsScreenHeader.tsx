import React from 'react';
import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
import FeatherIcon from 'react-native-vector-icons/Feather';

const CoinsScreenHeader = ({
  currency,
  leftTabAction,
  rightTabAction,
}) => {
  return (
    <View style={styles.mainContainer}>
      <Text
        style={styles.currency}>
        {currency}
      </Text>
      <Text style={styles.title}>Coins</Text>
      <View
        style={styles.buttons}>
        <TouchableOpacity
          onPress={leftTabAction}
          style={styles.button}>
          {/* <FeatherIcon color="#58AEE0" name={'bar-chart'} size={20} /> */}
          <Text
            style={styles.buttonTitle}>
            TOP 100
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={rightTabAction}
          style={styles.button}>
          <Text
            style={styles.buttonTitle}>
            MARKET CAP
          </Text>
          {/* <FeatherIcon color="#58AEE0" name={'sliders'} size={20} /> */}
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  mainContainer: {
    paddingVertical: 20,
    margin: 10
  },
  currency: {
    alignSelf: 'flex-end',
    fontSize: 18,
    color: '#58aee0',
    padding: 10,
  },
  title: {
    fontSize: 30,
    fontWeight: 'bold'
  },
  buttons: {
    flexDirection: 'row',
    paddingVertical: 10,
    justifyContent: 'space-between',
  },
  button: {
    backgroundColor: '#e4e6ea',
    paddingHorizontal: 15,
    paddingVertical: 10,
    borderWidth: 1,
    borderColor: 'transparent',
    borderRadius: 15,
    flexDirection: 'row',
  },
  buttonTitle: {
    color: '#58aee0',
    fontSize: 16,
    fontWeight: 'bold',
    paddingLeft: 10,
  }
})
export default CoinsScreenHeader;