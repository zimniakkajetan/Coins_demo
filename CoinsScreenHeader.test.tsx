import React from 'react';
import CoinsScreenHeader from './CoinsScreenHeader';


import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
      .create(<CoinsScreenHeader
        currency={'USD'}
        leftTabAction={() => {}}
        rightTabAction={() => {}}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

