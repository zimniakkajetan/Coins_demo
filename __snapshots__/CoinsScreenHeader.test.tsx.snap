// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza corretamente 1`] = `
<View
  style={
    Object {
      "margin": 10,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-end",
        "color": "#58aee0",
        "fontSize": 18,
        "padding": 10,
      }
    }
  >
    USD
  </Text>
  <Text
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    Coins
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e4e6ea",
          "borderColor": "transparent",
          "borderRadius": 15,
          "borderWidth": 1,
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#58aee0",
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingLeft": 10,
          }
        }
      >
        TOP 100
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e4e6ea",
          "borderColor": "transparent",
          "borderRadius": 15,
          "borderWidth": 1,
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#58aee0",
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingLeft": 10,
          }
        }
      >
        MARKET CAP
      </Text>
    </View>
  </View>
</View>
`;
